oteapi/models/genericconfig.py

```diff
@@ -0,0 +1,60 @@
+"""Generic data model for configuration attributes."""
+from typing import Any, Optional
+
+from pydantic import BaseModel, Field
+
+
+class AttrDict(BaseModel):
+    """An object whose attributes can also be accessed through
+    subscription, like with a dictionary."""
+
+    class Config:
+        """Class for configuration of pydantic models."""
+
+        extra = "allow"
+
+    def __contains__(self, name: Any) -> bool:
+        """Enable using the 'in' operator on this object."""
+        return self.__dict__.__contains__(name)
+
+    def __delitem__(self, key: str) -> None:
+        """Enable deletion access through subscription."""
+        if key in dir(self):
+            self.__delattr__(key)
```

Comment:
This is tricky. Since it's a pydantic model, there are some other internal `BaseModel` attributes that should be updated alongside this. I'm unsure whether calling `__delattr__` will update these.
The ones I'm thinking of are `__fields__` and `__fields_set__` (or something to that effect).

This might be fine, but having a test for these methods is crucial, to ensure they properly update the object as is intended.

Reply:
Thanks for the hint. I've updated the relevant attributes now. Now `del AttrDict["name"]` removes `name` from `AttrDict.__fields__` and `AttrDict.__fields_set__` (while `del AttrDict.name` doesn't) I plan on adding unit test(s) for the `Config` classes as well, if time allows for it.

Reply:
Would be great with the tests.
Just testing that the expected behavior is kept is enough. I can't speak for what people would expect when using `del`, but feel free to define that and state it in the tests (and PR). Then it is easier to spot for the reviewer and comment on eventually.